### PR TITLE
[GCI] #2 Change recipes to download safe sources

### PR DESCRIPTION
**Pull request by cyj0912**
Please check my work. Thank you.

### CODE REVIEW

**Comment by humdingerb** [comment]
Check out those errors in the Travis CI build. Once those are fixed, please squach your commits, see the last topic on https://github.com/haikuports/haikuports/wiki/DevelopmentModel

**Comment by humdingerb** [comment]
I'm sorry I've forgot to mention it before, can you please adjust the ordering of the blocks etc. mentioned in the wiki [Guidelines](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines).

**Comment by humdingerb** [comment]
Please have another look at ordering and the rest of the style (blank lines, 80char limit, etc.). This is a template:
```
SUMMARY
DESCRIPTION="Description, lines are \
max. 80 chars long."
HOMEPAGE
COPYRIGHT="Thing one
	Thing two"
LICENSE="MIT
	GNU GPL v2"
REVISION
SOURCE_URI
CHECKSUM_SHA256
SOURCE_DIR
PATCHES
	{blankline}
ARCHITECTURES
SECONDARY_ARCHITECTURES
	{blankline}
PROVIDES="
	application = $portVersion
	cmd:application
	"
REQUIRES
	{blankline}
PROVIDES_devel
REQUIRES_devel
	{blankline}
BUILD_REQUIRES
BUILD_PREREQUIRES
	{blankline}
BUILD()
	{blankline}
INSTALL
	{blankline}
TEST
```

**Comment by cyj0912** [comment]
Sorry I just realized a serious problem regarding the checksums. There are some recipes that were using the master branch, and hence I make them to download a tarball of the master branch. However the checksum changes whenever the master branch gets updated. What's the policy that I would follow to fix this?

**Comment by mmuman** [comment]
In this case you probably want to specify a specific commit for the tarball, so it's always the same (I hope, not sure that's always the case).

**Comment by pulkomandy** [comment]
Yes, in case of github you'll want to point either to a "release" archive (see the releases tab l in github), or if there is no matching release, to the archive related to a specific commit.

File: dev-embedded/arduino/arduino-1.7.7.recipe [hunk 1]
```diff
@@ -3,32 +3,34 @@ DESCRIPTION="Arduino is an open-source tool used for building electronics projec
...
 HOMEPAGE="http://arduino.org"
-LICENSE="GNU GPL v2"
 COPYRIGHT="2015 Arduino S.r.l. - Italy"
-
+LICENSE="GNU GPL v2"
+REVISION="1"
 SOURCE_URI="http://download.arduino.org/IDE/1.7.7/arduino-1.7.7.org-linux32.tar.xz"
+SOURCE_URI_2="https://github.com/scream3r/java-simple-serial-connector/archive/f5564869f8a70503c29c0d24609245321ba39b2c.tar.gz"
+SOURCE_URI_3="https://github.com/arduino-org/Arduino/archive/67152784328dd589ad34c9aeae54158a2b92a89d.tar.gz"
+SOURCE_URI_4="https://github.com/arduino-org/Arduino/archive/67152784328dd589ad34c9aeae54158a2b92a89d.tar.gz"
```
Comment (pulkomandy) [inline dev-embedded/arduino/arduino-1.7.7.recipe]: Is it ok that sources 3 and 4 are the same link?

**Comment by cyj0912** [comment]
As far as my knowledge, the haikuporter system would not download the same tarball a second time, it would simply extract the previously downloaded one. The way this recipe is set up is that 2 sub-projects within the same tarball need to be built. I can't think of a better way to write this. Please inform me of a better way if there is one. Thank you.

**Pull request merged**